github-actions/create-pr-for-changes/lib/main.ts

```diff
@@ -92,10 +92,34 @@ async function main(): Promise<void> {
 
     if (matchingPrs.length > 0) {
       // A PR for the same set of changes does already exist. Exit.
-      core.info(
-        `Skipping PR creation, because there is already a PR: #${matchingPrs[0].number} ` +
-          `(${matchingPrs[0].html_url})`,
-      );
+      for (const matchingPr of matchingPrs) {
+        // Check whether there is a PR is conflicting
+        const {
+          data: {mergeable},
+        } = await git.github.pulls.get({
+          owner: repo.owner,
+          repo: repo.name,
+          pull_number: matchingPr.number,
+        });
+
+        core.info(
+          `Skipping PR creation, because there is already a PR: #${matchingPr.number} ` +
+            `(${matchingPr.html_url})`,
+        );
+
+        if (!mergeable) {
+          core.info(`PR is not mergable, rebasing branch: #${branchName}`);
+          // Rebase PR
+          git.run(['checkout', '-b', branchName]);
```

Comment:
I don't quite follow. How does this rebase? do I miss something?

Reply:
is this just because we re-push the currently staged changes and override the existing PR? Might be worth adding that to the comment

Reply:
Yes